### ISSUE TITLE
Notebook run action not visible on Windows but visible on Mac

### ISSUE THREAD

**Issue by misolori** [microsoft/vscode-jupyter#6017]
Trying to figure if this is native notebook bug or a jupyter extension bug. I'm currently trying capture the end to end experience for opening a notebook file in VS Code.

![image](https://user-images.githubusercontent.com/35271042/119859850-9bb60280-beca-11eb-9c22-d17d66f0e049.png)

1. Open VS Code with an empty data & extensions directory
2. Open a notebook file
3. Install the recommended extension (Jupyter)
4. Re-open file
5. Trust notebook
6. :bug: Run a cell, run action is not visible

However on macOS, the run actions DOES show up, so there is some weird behavior here:

![image](https://user-images.githubusercontent.com/35271042/119859892-a7092e00-beca-11eb-8ac5-86e7cc3e668d.png)

Python     2.7.15
Edition	Windows 10 Pro
Version	20H2
Installed on	‎3/‎24/‎2021
OS build	19042.928
Experience	Windows Feature Experience Pack 120.2212.551.0

Version: 1.57.0-insider (user setup)
Commit: 1082913dd012e21cc5f35a06f15c480ca447f67f
Date: 2021-05-27T06:55:24.593Z
Electron: 12.0.9
Chrome: 89.0.4389.128
Node.js: 14.16.0
V8: 8.9.255.25-electron.0
OS: Windows_NT x64 10.0.19042

cc @claudiaregio

**Comment by joyceerhl** [comment]
I can reproduce the exact behavior that Miguel describes if I uninstall all Python interpreters, uninstall the Python extension, and delete all kernels from my jupyter/kernels directory.

@misolori You mentioned you have the default system Python 2.7, I assume that is on Mac? Do you have Python installed on your Windows machine at all?

If you don't have Python installed on your Windows machine (and no kernels), I think what might be happening here is that if you have no Python interpreters or kernels, the Jupyter extension does not register any notebook controller, and under the kernel push UX, the run button only appears when a kernel becomes available. If the Jupyter extension can't create and return any controllers, the run button stays hidden. On macOS Python 2.7 is available by default so we are able to find and set up a controller for that. We have plans to work around this by returning a dummy Python 3 kernel which would address this bug: https://github.com/microsoft/vscode-jupyter/issues/5861

**Comment by misolori** [comment]
> @misolori You mentioned you have the default system Python 2.7, I assume that is on Mac? Do you have Python installed on your Windows machine at all?

I misspoke, I had previously installed Python 2.7 so I did have that installed. I also tried the repro steps on a Windows Sandbox environment and still saw the same issue. However, in today's insiders I am no longer seeing this issue so not sure if the Jupyter extension pushed an update that fixed this. Here are my current versions:

Jupyter: v2021.6.885080120
Version: 1.57.0-insider (user setup)
Commit: 541a5d4fa70ab3095f23b28b88914e1f6a8f65c4
Date: 2021-05-28T09:50:57.063Z
Electron: 12.0.9
Chrome: 89.0.4389.128
Node.js: 14.16.0
V8: 8.9.255.25-electron.0
OS: Windows_NT x64 10.0.19041
